metricflow/sql_clients/big_query.py

```diff
@@ -56,12 +57,31 @@ def __init__(self, project_id: str = "", password: Optional[str] = None, query_s
         # Without pool_pre_ping, it's possible for timed-out connections to be returned to the client and cause errors.
         # However, this can cause increase latency for slow engines.
         self._password = password
+        self._project_id = project_id
         super().__init__(
             engine=BigQuerySqlClient._create_bq_engine(
                 query_string=query_string, project_id=project_id, password=password
             )
         )
 
+    def _engine_specific_explain_implementation(
+        self, stmt: str, bind_params: SqlBindParameters
+    ) -> Optional[List[Exception]]:
+        """Overrides base `_engine_specific_explain_implementation` function for BigQuery specifics"""
+        _engine = self._create_bq_engine(
+            query_string="dry_run=true", project_id=self._project_id, password=self._password
+        )
+        exceptions: List[Exception] = []
+        with _engine.connect() as conn:
+            for s in stmt.split(";"):
```

Comment:
This is buggy - `;` can be in a string literal. Ditto for the other similar call site. Is this splitting necessary?

Reply:
Doing some research we could use [sqlparse](https://pypi.org/project/sqlparse/) so that we can use `sqlparse.split()`

Reply:
Paul and I talked offline. We don't want to encourage multiple statements in a statement to be queried / executed / explained. MetricFlow does not currently do these kind of compound statements. Thus we're going to drop the splitting.